sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobAsyncClient.java

```diff
@@ -253,19 +251,21 @@ as we can guarantee we only need at most two buffers for any call to write (two
             buffer filling up with more data to write). We use flatMapSequential because we need to guarantee we
             preserve the ordering of the buffers, but we don't really care if one is split before another.
              */
-            Flux<ByteBuffer> chunkedSource = data.flatMapSequential(buffer -> {
-                if (buffer.remaining() <= blockSize) {
-                    return Flux.just(buffer);
-                }
-                int numSplits = (int) Math.ceil(buffer.remaining() / (double) blockSize);
-                return Flux.range(0, numSplits)
-                    .map(i -> {
-                        ByteBuffer duplicate = buffer.duplicate().asReadOnlyBuffer();
-                        duplicate.position(i * blockSize);
-                        duplicate.limit(Math.min(duplicate.limit(), (i + 1) * blockSize));
-                        return duplicate;
-                    });
-            });
+            Flux<ByteBuffer> chunkedSource = data
+                .filter(buffer -> buffer.remaining() > 0)
```

Comment:
Would this be better?

```suggestion
                .filter(buffer -> buffer.hasRemaining())
```